src/gtk/webview_webkit2.cpp

```diff
@@ -1092,14 +1097,107 @@ wxString wxWebViewWebKit::GetPageText() const
     }
     return wxString();
 }
+static void wxgtk_run_javascript_cb(WebKitWebView *,
+                                    GAsyncResult *res,
+                                    GAsyncResult **res_out)
+{
+    *res_out = (GAsyncResult*)g_object_ref(res);
+}
+
+bool JSResultToString(GObject *object, GAsyncResult *result, wxString* output)
+{
+    WebKitJavascriptResult  *js_result;
```

Comment:
Prefer to declare the variable when initializing it, this is C++, not C. I.e. write

    WebKitJavascriptResult* js_result = webkit_web_view_run_javascript_finish(...);